src/main/java/dao/ReaderImpl.java

```diff
@@ -0,0 +1,29 @@
+package dao;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import validator.Validator;
+import validator.ValidatorImpl;
+
+public class ReaderImpl implements Reader {
+    private static final Validator validator = new ValidatorImpl();
+
+    @Override
+    public List<String> reader(String fileName) {
```

Comment:
it's better to divide validation from reading, this way we will be able to reuse our reader whener we need to read from file in other parts of an application